### ISSUE TITLE
Deleting a thing leaves references intact

### ISSUE THREAD

**Issue by fefi42** [semi-technologies/weaviate#1186]
When deleting a referenced thing the reference to that thing will not be deleted. 

Question should the reference be deleted?

Issue: The reference is shown as `null` in the GraphQL endpoint so as if it does not exist.
Only the rest endpoint shows the reference actually still persists.

Reproduction:
``` python
import weaviate
import time

client = weaviate.Client("http://localhost:8080")

client.create_schema("https://raw.githubusercontent.com/semi-technologies/weaviate-python-client/master/documentation/getting_started/people_schema.json")


client.create_thing({"name": "Legends"}, "Group", "2db436b5-0557-5016-9c5f-531412adf9c6")
client.create_thing({"name": "Alan Turing"}, "Person", "1c9cd584-88fe-5010-83d0-017cb3fcb446")

time.sleep(2)

client.add_reference_to_thing("2db436b5-0557-5016-9c5f-531412adf9c6", "members", "1c9cd584-88fe-5010-83d0-017cb3fcb446")

client.delete_thing("1c9cd584-88fe-5010-83d0-017cb3fcb446")
```

In this example the thing `2db436b5-0557-5016-9c5f-531412adf9c6` will show no references as `Members` however the reference still exists.

**Comment by etiennedi** [comment]
> Question should the reference be deleted?

I see two options:
1. Deleting should be blocked if an object is a target of a reference (like a foreign key id in a RDBMS)
2. Deleting should be a cascading delete and also clean up all references. This could potentially be problematic on a massive dataset. Imagine you have a twitter follower network with 1bn followers of a high-profile politician. If we were to delete the politican's twitter account, we'd have to alter 1bn objects, which isn't feasible. On a small dataset this might be a nice convenience option though.

**Comment by bobvanluijt** [comment]
Could it be set as a flag?

- Defaults to option 1
- `?force=true` deletes using the cascading delete scheme

**Comment by etiennedi** [comment]
Yes, I think this would be an option. And `?force=true` would fail if `n=>10000` as that's the internal query limit of the current db anyway.

**Comment by fefi42** [comment]
I like the flag solution.

Can we also solve that the reference is shown as null in GraphQL while it still exits?

**Comment by bobvanluijt** [comment]
> Can we also solve that the reference is shown as null in GraphQL while it still exits?

Maybe combine with #1163?

**Comment by etiennedi** [comment]
@fefi42 

> Can we also solve that the reference is shown as null in GraphQL while it still exits?

What would be the correct thing to show in your opinion?

**Comment by fefi42** [comment]
Good question currently it is not distinguishable if there is no reference or there is a reference not leading anywhere.
Maybe instead of`"Members": null` we can write it in an array:
```
"Members": [
    {
    null
    }
],
```
This probably also has not the best usability because you have to know what the difference is. But it would make it at least distinguishable.
